src/frontend/core/executor/impl/TriangleCountExecutor.cpp

```diff
@@ -0,0 +1,1560 @@
+
+        for (partitionIterator = partitionList.begin(); partitionIterator != partitionList.end(); ++partitionIterator) {
+            partitionCount++;
+            int workerPort = atoi(string(currentWorker.port).c_str());
+            int workerDataPort = atoi(string(currentWorker.dataPort).c_str());
+            triangleCount_logger.log("====>PORT:" + std::to_string(workerPort), "info");
+
+            partitionId = *partitionIterator;
+            intermRes.push_back(
+                    std::async(std::launch::async, TriangleCountExecutor::getTriangleCount, atoi(graphId.c_str()), host,
+                               workerPort, workerDataPort, atoi(partitionId.c_str()), masterIP, uniqueId,
+                               isCompositeAggregation, threadPriority));
+        }
+    }
+
+    PerformanceUtil performanceUtil;
+    performanceUtil.init();
+
+    std::string query = "SELECT attempt from graph_sla INNER JOIN sla_category where graph_sla.id_sla_category=sla_category.id and "
+                        "graph_sla.graph_id='" + graphId + "' and graph_sla.partition_count='" + std::to_string(partitionCount) +
+                        "' and sla_category.category='" + Conts::SLA_CATEGORY::LATENCY + "';";
+
+    std::vector<vector<pair<string, string>>> queryResults = perfDB.runSelect(query);
+
+    if (queryResults.size() > 0) {
+        std::string attemptString = queryResults[0][0].second;
+        int calibratedAttempts = atoi(attemptString.c_str());
+
+        if (calibratedAttempts >= Conts::MAX_SLA_CALIBRATE_ATTEMPTS) {
+            canCalibrate = "false";
+        }
+    }
+
+    while (!workerResponded && canCalibrate == "true") {
+        performanceUtil.collectSLAResourceConsumption(graphId,TRIANGLES,Conts::SLA_CATEGORY::LATENCY,slaStatCount,partitionCount);
+        slaStatCount++;
+        sleep(5);
+    }
+
+    for (auto &&futureCall:intermRes) {
+        result += futureCall.get();
+    }
+
+    if (!isCompositeAggregation) {
+        long aggregatedTriangleCount = TriangleCountExecutor::aggregateCentralStoreTriangles(sqlite, graphId, masterIP,
+                                                                                             threadPriority);
+        result += aggregatedTriangleCount;
+        triangleCount_logger.log("###TRIANGLE-COUNT-EXECUTOR### Getting Triangle Count : Completed: Triangles " + to_string(result),
+                            "info");
+        JobResponse jobResponse;
+        jobResponse.setJobId(request.getJobId());
+        jobResponse.addParameter(Conts::PARAM_KEYS::TRIANGLE_COUNT, std::to_string(result));
+        responseVector.push_back(jobResponse);
+        responseMap[request.getJobId()] = jobResponse;
+    }
+
+    auto end = chrono::high_resolution_clock::now();
+    auto dur = end - begin;
+    auto msDuration = std::chrono::duration_cast<std::chrono::milliseconds>(dur).count();
+
+    std::string durationString = std::to_string(msDuration);
+
+    if (canCalibrate == "true") {
```

Comment:
We should change this to `canCalibrate.compare("true") == 0`

Reply:
Changed this parameter to boolean